cache4k/src/commonMain/kotlin/io/github/reactivecircus/cache4k/FakeTimeSource.kt

```diff
@@ -28,21 +29,24 @@ public class FakeTimeSource : AbstractLongTimeSource(unit = DurationUnit.NANOSEC
     public operator fun plusAssign(duration: Duration) {
         val delta = duration.toDouble(unit)
         val longDelta = delta.toLong()
-        reading.set(
-            reading.get().let { currentReading ->
+        reading.value =
+            reading.value.let { currentReading ->
```

Comment:
Should we use [`update(function: (T) -> T)`](https://github.com/Kotlin/kotlinx.atomicfu/blob/450b056efa4c8e4f922f2a3f8d1958404513b1f7/atomicfu/src/commonMain/kotlin/kotlinx/atomicfu/AtomicFU.common.kt#L163) instead?

It would also address this from the [atomicfu dos and don'ts](https://github.com/Kotlin/kotlinx.atomicfu#dos-and-donts):

> Do not introduce complex data flow in parameters to atomic variable operations, i.e. top.value = complex_expression and top.compareAndSet(cur, complex_expression) are not supported (more specifically, complex_expression should not have branches in its compiled representation). Extract complex_expression into a variable when needed.